bundles/org.openhab.binding.senechome/src/main/java/org/openhab/binding/senechome/internal/SenecHomeHandler.java

```diff
@@ -0,0 +1,316 @@
+                        new QuantityType<Dimensionless>(pvLimitation, SmartHomeUnits.PERCENT));
+            }
+
+            Channel channelLimitationState = getThing()
+                    .getChannel(SenecHomeBindingConstants.CHANNEL_SENEC_POWER_LIMITATION_STATE);
+            if (channelLimitationState != null) {
+                updatePowerLimitationStatus(channelLimitationState,
+                        (100 - pvLimitation.intValue()) <= config.limitationTresholdValue, config.limitationDuration);
+            }
+
+            Channel channelConsumption = getThing()
+                    .getChannel(SenecHomeBindingConstants.CHANNEL_SENEC_POWER_CONSUMPTION);
+            if (channelConsumption != null) {
+                updateState(channelConsumption.getUID(),
+                        new QuantityType<Power>(
+                                getSenecValue(response.energy.homePowerConsumption).setScale(2, RoundingMode.HALF_UP),
+                                SmartHomeUnits.WATT));
+            }
+
+            Channel channelEnergyProduction = getThing()
+                    .getChannel(SenecHomeBindingConstants.CHANNEL_SENEC_ENERGY_PRODUCTION);
+            if (channelEnergyProduction != null) {
+                updateState(channelEnergyProduction.getUID(), new QuantityType<Power>(
+                        getSenecValue(response.energy.inverterPowerGeneration).setScale(0, RoundingMode.HALF_UP),
+                        SmartHomeUnits.WATT));
+            }
+
+            Channel channelBatteryPower = getThing().getChannel(SenecHomeBindingConstants.CHANNEL_SENEC_BATTERY_POWER);
+            if (channelBatteryPower != null) {
+                updateState(channelBatteryPower.getUID(),
+                        new QuantityType<Power>(
+                                getSenecValue(response.energy.batteryPower).setScale(2, RoundingMode.HALF_UP),
+                                SmartHomeUnits.WATT));
+            }
+
+            Channel channelBatteryFuelCharge = getThing()
+                    .getChannel(SenecHomeBindingConstants.CHANNEL_SENEC_BATTERY_FUEL_CHARGE);
+            if (channelBatteryFuelCharge != null) {
+                updateState(channelBatteryFuelCharge.getUID(),
+                        new QuantityType<Dimensionless>(
+                                getSenecValue(response.energy.batteryFuelCharge).setScale(0, RoundingMode.HALF_UP),
+                                SmartHomeUnits.PERCENT));
+            }
+
+            Channel channelBatteryState = getThing().getChannel(SenecHomeBindingConstants.CHANNEL_SENEC_BATTERY_STATE);
+            if (channelBatteryState != null) {
+                updateBatteryState(channelBatteryState, getSenecValue(response.energy.batteryState).intValue());
+            }
+
+            updateGridPowerValues(getThing(), getSenecValue(response.grid.currentGridValue));
+
+            updateStatus(ThingStatus.ONLINE);
+        } catch (Exception e) {
```

Comment:
Please do not catch `Exception`, be more specific. `Exception` includes NPE and thus hides programming errors